force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,149 @@
+                mails.add(mail);
+            }        
+            
+            try {
+                Messaging.sendEmail(mails);
+            } catch (Exception e){
+                System.debug('Exception: ' + e.getMessage());
+            }
+        }
+    // assignPrimaryContact:
+    // - Assigns a primary contact with the title of 'VP Sales' to undeleted Opportunities.
+    // - Only updates the Opportunities that don't already have a primary contact.
+    // */
+    public static void assignPrimaryContact(List<Opportunity> oppList, Map<Id, Opportunity> oppMap) {        
+        Map<Id, Opportunity> oppMapToUpdate = new Map<Id, Opportunity>();
+        Set<Id> accountIds = new Set<Id>();
+            for (Opportunity opp : oppMap.values()){
+            if(opp.Primary_Contact__c == null && opp.AccountId != null){
+                accountIds.add(opp.AccountId);
+            }
+        }
+        Map<Id, Contact> primaryContactMap = new Map<Id, Contact>();     
+        for (Contact contact : [SELECT Id, AccountId FROM Contact WHERE Title = 'VP Sales' AND AccountId IN :accountIds])  {
+            primaryContactMap.put(contact.AccountId, contact);
+        } 
+
+        for (Opportunity opp : oppMap.values()){            
+            if (opp.Primary_Contact__c == null){
+                Opportunity oppToUpdate = new Opportunity(Id = opp.Id);
+                oppToUpdate.Primary_Contact__c = primaryContactMap.get(opp.AccountId).Id;
+                oppMapToUpdate.put(opp.Id, oppToUpdate);
+            }
+        }
+        update oppMapToUpdate.values();
+    }
+
+    public static void amountValidation(List<Opportunity> oppList){
+        for(Opportunity opp : oppList){
+            if(opp.Amount <= 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+                        
+            }
+
+        }
+    }
+
+    public static void setPrimaryContact(List<Opportunity> oppList){
+        List<Id> accountIds = new List<Id>();
+            for (Opportunity opp : oppList) {
+                if (opp.AccountId != null) {
+                    accountIds.add(opp.AccountId);
+                }
+            }
+            
+            Map<Id, Contact> ceoContactsByAccount = new Map<Id, Contact> ([SELECT Id, AccountId, Title FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO']);
+           
+            Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+        
+                for (Contact cont : ceoContactsByAccount.values()) {
+                    accountIdToContact.put(cont.AccountId, cont);
+                }
+                
+            for(Opportunity opp : oppList){
+                if(opp.Primary_Contact__c == null && accountIdToContact.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContact.get(opp.AccountId).Id;
+                }
+            }
+        }
```

Comment:
Does this method look very similar to the one above? This is where you should be looking to combine functionality and not repeat code. Can you combine them into 1 method by dynamically passing in what is different about them?